### ISSUE TITLE
Listing conversation with only order/sort fails unless specifying a user or admin

### ISSUE THREAD

**Issue by thewheat** [intercom/intercom-java#163]
- Via the API it is possible to list conversations by via a GET to the `/conversations` endpoint with `sort` or `order` parameters
- If you try this via this library you get an error: `a user or admin type must be supplied for a conversation query`

**API docs**
https://developers.intercom.com/reference#section-all-conversations
![image](https://user-images.githubusercontent.com/892961/33612733-820b0fd8-da0d-11e7-9dd6-24c77d190f2c.png)

**Test**
![image](https://user-images.githubusercontent.com/892961/33154882-79532c04-d026-11e7-9268-7deb3de8c640.png)

**Reason for failture**
This validation currently prevents listing by conversations https://github.com/intercom/intercom-java/blob/71439e56a01c73b03c54d427fc60d7a0464f62f0/intercom-java/src/main/java/io/intercom/api/Conversation.java#L144-L146

## Version info
  - intercom-java version: 2.2.11
  - Java version: 1.8.0_101

## Expected behavior
List all conversations

## Actual behavior
Exception and no API call made

## Steps to reproduce
 1. Run code with the following code
```
Map<String, String> params = Maps.newHashMap();
params.put("order", "created_at");
ConversationCollection conversations = Conversation.list(params);
```

## Logs
```
Exception in thread "main" io.intercom.api.InvalidException: a user or admin type must be supplied for a conversation query
	at io.intercom.api.Conversation.validateListRequest(Conversation.java:145)
	at io.intercom.api.Conversation.list(Conversation.java:45)
	at org.thewheatfield.intercom.java.example.Main.main(Main.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

**Comment by mmartinic** [comment]
We can use `Conversation.list()` but it's undocumented. I'll add an example to readme

**Comment by thewheat** [comment]
Ah I worded this issue badly @mmartinic The sample code is what needs supporting where you do a listing with a specific sort/order value

```
Map<String, String> params = Maps.newHashMap();
params.put("order", "created_at");
ConversationCollection conversations = Conversation.list(params);
```

When making that call, it goes to the `Conversation.list(params)` function which has extra validation 

https://github.com/intercom/intercom-java/blob/fe3accf876fad1a3e3d8e9de49a87ce7a965035b/intercom-java/src/main/java/io/intercom/api/Conversation.java#L44-L48


I'll reopen the issue and rename the title and clarify this a bit more in the original report

**Comment by thewheat** [comment]
This is fixed with the latest 2.2.12 release 👍   https://github.com/intercom/intercom-java/releases/tag/2.2.12
